src/features/Instructors/InstructorsPage/index.jsx

```diff
@@ -21,13 +21,19 @@ const InstructorsPage = () => {
 
   useEffect(() => {
     if (Object.keys(selectedInstitution).length > 0) {
-      dispatch(fetchInstructorsData(selectedInstitution.id, currentPage, stateInstructors.filters));
+      dispatch(fetchInstructorsData(selectedInstitution.id, initialPage));
     }
 
     return () => {
       dispatch(resetInstructorsRequest());
     };
-  }, [selectedInstitution, dispatch, currentPage]); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [selectedInstitution, dispatch]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    if (Object.keys(selectedInstitution).length > 0) {
+      dispatch(fetchInstructorsData(selectedInstitution.id, currentPage, stateInstructors.filters));
+    }
+  }, [currentPage]); // eslint-disable-line react-hooks/exhaustive-deps
 
```

Comment:
@01001110J I tried to use only one, but if you apply filters then you go to other page and then come back to Instructors page, the filters will be applied yet. 

So, the second useEffect hook is to avoid that scenario and clean the previous result.

Reply:
> the filters will be applied yet

If we do not want to preserve the filters once the user leave the page, why you just clean it once the component is being removed, doing something like:

```js
useEffect(() => {
    if (Object.keys(selectedInstitution).length > 0) {
      dispatch(fetchInstructorsData(selectedInstitution.id, currentPage, filters));
    }

    return () => {
      dispatch(resetInstructorsRequest());
      // This should clean the filter selection
      // once the component is removed from the view
     dispatch(updateFilters({})) 
    };
  }, [selectedInstitution, dispatch, currentPage, filters]);
```

Reply:
@01001110J excellent. I skipped filters in the dependencies because If I add it then it fails. It keeps loading in a loop and never stop.